adscompstat/tasks.py

```diff
@@ -453,3 +459,49 @@ def task_export_completeness_to_json():
                 )
         except Exception as err:
             logger.error("Unable to export completeness data to disk: %s" % err)
+
+
+@app.task(queue="get-logfiles")
+def task_retry_mismatched():
+    with app.session_scope() as session:
+        batch_count = app.conf.get("RECORDS_PER_BATCH", 100)
+        try:
+            result = (
+                session.query(master.harvest_filepath).filter(master.matchtype == "mismatch").all()
+            )
+            batch = []
+            for r in result:
+                batch.append(r[0])
+                if len(batch) == batch_count:
+                    logger.debug("Calling task_process_meta with batch '%s'" % batch)
+                    task_process_meta.delay(batch)
+                    batch = []
+            if len(batch):
+                logger.debug("Calling task_process_meta with batch '%s'" % batch)
+                task_process_meta.delay(batch)
+        except Exception as err:
+            logger.warning("Error reprocessing mismatched records: %s" % err)
+
+
+@app.task(queue="get-logfiles")
+def task_retry_unmatched():
+    with app.session_scope() as session:
+        batch_count = app.conf.get("RECORDS_PER_BATCH", 100)
+        try:
+            result = (
+                session.query(master.harvest_filepath)
+                .filter(master.matchtype == "unmatched")
+                .all()
+            )
+            batch = []
+            for r in result:
+                batch.append(r[0])
+                if len(batch) == batch_count:
+                    logger.debug("Calling task_process_meta with batch '%s'" % batch)
+                    task_process_meta.delay(batch)
+                    batch = []
+            if len(batch):
+                logger.debug("Calling task_process_meta with batch '%s'" % batch)
+                task_process_meta.delay(batch)
+        except Exception as err:
+            logger.warning("Error reprocessing unmatched records: %s" % err)
```

Comment:
It looks like the functions `task_retry_mismatched` and `task_retry_unmatched` can be made into one function where you specify the "matchtype" as argument? Is there reason why it's better to keep two separate functions?

Reply:
This is a great idea.  I changed this to accept a string containing the record's `matchtype`, and changed `run.py` so that calling this option will iteratively call this task with the strings `mismatch`, `unmatched`, and `failed`.  `Failed` was not in the original PR but should be, if (for example) a bugfix is made to `adsingestp.parsers.crossref` that will address reasons why the records failed.